src/thin_client.rs

```diff
@@ -238,6 +238,32 @@ impl ThinClient {
         0
     }
 
+    /// Request the last Entry ID from the server without blocking.
+    /// Returns the last_id Hash or None if there was no response from the server.
+    pub fn try_get_last_id(&mut self, mut num_retries: u64) -> Option<Hash> {
```

Comment:
Can the get_last_id function call this one with num_retries with a high value?

Reply:
I'm okay with doing that but I didn't want to mess with the permanent blocking nature of that fn.

Reply:
@sakridge how about making get_last_id call this in a loop (with like 10 retries per iteration)?

Reply:
That's fine too.